File: abc274/d.cpp
#include <bits/stdc++.h>
#define ll long long
using namespace std;

int coordinate_to_idx(int c)
{
    return c + 10000;
}

int main()
{
    int N, x, y;
    cin >> N >> x >> y;
    vector<int> A(N);
    for (int i = 0; i < N; i++)
    {
        cin >> A[i];
    }
    vector<int> A_odd(N / 2 + N % 2); // xを作る
    vector<int> A_even(N / 2);        // yを作る
    for (int i = 0; i < N; i++)
    {
        if (i % 2 == 0)
        {
            A_odd[i / 2] = A[i];
        }
        if (i % 2 == 1)
        {
            A_even[i / 2 - 1] = A[i];
        }
    }
    vector<vector<bool>> x_dp(A_odd.size() + 1, vector<bool>(20010));
    vector<vector<bool>> y_dp(A_even.size() + 1, vector<bool>(20010));

    cout << x_dp.size() << ","  << y_dp.size() << endl;

    x_dp[0][coordinate_to_idx(0)] = true;
    x_dp[1][coordinate_to_idx(A_odd[0])] = true;
    y_dp[0][coordinate_to_idx(0)] = true;

    for (unsigned long i = 0; i < A_odd.size(); i++)
    {
        for (int j = -10000; j < 100001; j++)
        {
            if (j - A_odd[i] >= -10000)
            {
                x_dp[i + 1][coordinate_to_idx(j)] = x_dp[i][coordinate_to_idx(j - A_odd[i])] || x_dp[i + 1][coordinate_to_idx(j)];
            }
            if (j + A_odd[i] <= 10000)
            {
                x_dp[i + 1][coordinate_to_idx(j)] = x_dp[i][coordinate_to_idx(j + A_odd[i])] || x_dp[i + 1][coordinate_to_idx(j)];
            }
        }
    }
    for (unsigned long i = 0; i < A_even.size(); i++)
    {
        for (int j = -10000; j < 100001; j++)
        {
            if (j - A_even[i] >= -10000)
            {
                y_dp[i + 1][coordinate_to_idx(j)] = y_dp[i][coordinate_to_idx(j - A_even[i])] || y_dp[i + 1][coordinate_to_idx(j)];
            }
            if (j + A_even[i] <= 10000)
            {
                y_dp[i + 1][coordinate_to_idx(j)] = y_dp[i][coordinate_to_idx(j + A_even[i])] || y_dp[i + 1][coordinate_to_idx(j)];
            }
        }
    }

    if (x_dp[x_dp.size() - 1][coordinate_to_idx(x)] && y_dp[y_dp.size() - 1][coordinate_to_idx(y)])
    {
        cout << "Yes" << endl;
    }
    else
    {
        cout << "No" << endl;
    }
}

File: abc274/d.py
N,x,y = map(int,input().split())
A = list(map(int,input().split()))
A_odd = [A[i] for i in range(N) if i%2==0] # 0からxを作る
A_even = [A[i] for i in range(N) if i%2==1] # 0からyを作る

def coordinate_to_dpidx(c:int):
  return c + 10000

x_dp = [[False for _ in range(20001)] for _ in range(len(A_odd)+1)]
y_dp = [[False for _ in range(20001)] for _ in range(len(A_even)+1)]

x_dp[0][coordinate_to_dpidx(0)] = True
x_dp[1][coordinate_to_dpidx(A_odd[0])] = True
y_dp[0][coordinate_to_dpidx(0)] = True


# x_dp
for i in range(2,len(A_odd)+1):
  for x_idx in range(-10000,10001):
    if x_idx-A_odd[i-1] >= -10000:  
      x_dp[i][coordinate_to_dpidx(x_idx)] = x_dp[i-1][coordinate_to_dpidx(x_idx-A_odd[i-1])] or x_dp[i][coordinate_to_dpidx(x_idx)]
    if x_idx+A_odd[i-1] <= 10000:
      x_dp[i][coordinate_to_dpidx(x_idx)] = x_dp[i-1][coordinate_to_dpidx(x_idx+A_odd[i-1])] or x_dp[i][coordinate_to_dpidx(x_idx)]

# print([[i-10000 for i, x in enumerate(x_dp[j]) if x == True] for j in range(len(x_dp))])
# y_dp
for i in range(1,len(A_even)+1):
  for y_idx in range(-10000,10000+1):
    if y_idx-A_even[i-1] >= -10000:  
      y_dp[i][coordinate_to_dpidx(y_idx)] = y_dp[i-1][coordinate_to_dpidx(y_idx-A_even[i-1])] or y_dp[i][coordinate_to_dpidx(y_idx)]
    if y_idx+A_even[i-1] <= 10000:
      y_dp[i][coordinate_to_dpidx(y_idx)] = y_dp[i-1][coordinate_to_dpidx(y_idx+A_even[i-1])] or y_dp[i][coordinate_to_dpidx(y_idx)]

# print([[i-10000 for i, x in enumerate(x_dp[j]) if x == True] for j in range(len(x_dp))])
if x_dp[len(x_dp)-1][coordinate_to_dpidx(x)] and y_dp[len(y_dp)-1][coordinate_to_dpidx(y)]:
    print("Yes")
else:
    print("No")



